src/containers/AuthorsList.js

```diff
@@ -1,14 +1,14 @@
 'use strict'
 
-import { LOADING_MORE_AUTHORS, NO_RESULT, PAGE_TITLE, REQUEST_PAGE_START_FROM } from '../constants/authors-list'
-import { fetchNextPageAuthors, sendSearchAuthors } from '../actions/authors'
+import { LOADING_MORE_AUTHORS, NO_RESULT, PAGE_TITLE, NUMBER_OF_FIRST_RESPONSE_PAGE } from '../constants/authors-list'
+import { searchAuthorsIfNeeded, setAuthorsListType } from '../actions/authors'
```

Comment:
Make `setAuthorsListType` as part of `state` of container.
Use `state.toShowAll`,`state.toShowSearchedAuthors` or `state. typeOfAuthorsListToRender ` to render the author list.